crypto/s2n_certificate.h

```diff
@@ -53,30 +53,26 @@ typedef enum {
     S2N_CERT_TYPE_ECDSA_FIXED_ECDH = 66,
 } s2n_cert_type;
 
-struct s2n_cert_public_key {
+struct s2n_cert {
```

Comment:
I've reverted the renaming. It still kind of nags me that the cert type is bundled up in a "key type", but I plan on addressing that plus some other refactoring in a separate PR.